torch/nn/modules/module.py

```diff
@@ -823,26 +831,48 @@ def compute_should_use_set_data(tensor, tensor_applied):
             # `with torch.no_grad():`
             with torch.no_grad():
                 param_applied = fn(param)
-            should_use_set_data = compute_should_use_set_data(param, param_applied)
-            if should_use_set_data:
-                param.data = param_applied
-                out_param = param
+            p_should_use_set_data = compute_should_use_set_data(param, param_applied)
+            p_should_use_swap_tensors = compute_should_use_swap_tensors(param)
+            # FIXME: This is wrong, param_grad bumps the use_count so we will never swap_tensors for grad
+            # nevertheless, passing param.grad to any function (even swap_tensors) bumps its use_count,
+            # so it seems like we are maybe doomed?
```

Comment:
@albanD 

I am running into an issue with the `use_count` check we added, this happens specifically for `.grad`.

Consider the following

```python
>>> import torch
>>> t = torch.randn(2, 3)
>>> p = torch.nn.Parameter(t)
>>> p.grad = torch.randn(2, 3)
>>> torch._C._tensor_use_count(p.grad._cdata)
1
>>> def foo(g):
>>>     print(torch._C._tensor_use_count(g._cdata)
>>> foo(p.grad) # 2
```